crates/wasm/src/wasm_planner.rs

```diff
@@ -0,0 +1,221 @@
+    ///     memo: `MemoPlaintext`
+    pub fn memo(&mut self, memo: JsValue) -> Result<(), Error> {
+        self.memo_inner(memo)?;
+        Ok(())
+    }
+
+    /// Add fee to plan
+    /// Arguments:
+    ///     fee: `Fee`
+    pub fn fee(&mut self, fee: JsValue) -> Result<(), Error> {
+        self.fee_inner(fee)?;
+        Ok(())
+    }
+
+    /// Add output to plan
+    /// Arguments:
+    ///     value: `Value`
+    ///     address: `Address`
+    pub fn output(&mut self, value: JsValue, address: JsValue) -> Result<(), Error> {
+        self.output_inner(value, address)?;
+        Ok(())
+    }
+
+    /// Add swap claim to plan
+    /// Arguments:
+    ///     swap_commitment: `StateCommitment`
+    #[wasm_bindgen(js_name = swapClaim)]
+    pub async fn swap_claim(&mut self, swap_commitment: JsValue) -> Result<(), Error> {
+        self.swap_claim_inner(swap_commitment).await?;
+        Ok(())
+    }
+
+    /// Add swap  to plan
+    /// Arguments:
+    ///     input_value: `Value`
+    ///     into_denom: `DenomMetadata`
+    ///     swap_claim_fee: `Fee`
+    ///     claim_address: `Address`
+    pub fn swap(
+        &mut self,
+        input_value: JsValue,
+        into_denom: JsValue,
+        swap_claim_fee: JsValue,
+        claim_address: JsValue,
+    ) -> Result<(), Error> {
+        self.swap_inner(input_value, into_denom, swap_claim_fee, claim_address)?;
+        Ok(())
+    }
+
+    /// Build transaction plan
+    /// Arguments:
+    ///     self_address: `Address`
+    /// Returns: `TransactionPlan`
+    pub async fn plan(&mut self, self_address: JsValue) -> Result<JsValue, Error> {
+        let plan = self.plan_inner(self_address).await?;
+        serde_wasm_bindgen::to_value(&plan)
+    }
+}
+
+impl WasmPlanner {
+    pub fn memo_inner(&mut self, memo: JsValue) -> WasmResult<()> {
+        let memo_proto: MemoPlaintext = serde_wasm_bindgen::from_value(memo)?;
+        let _ = self.planner.memo(memo_proto.try_into()?);
+        Ok(())
+    }
+
+    pub fn fee_inner(&mut self, fee: JsValue) -> WasmResult<()> {
+        let fee_proto: Fee = serde_wasm_bindgen::from_value(fee)?;
+
+        self.planner.fee(fee_proto.try_into()?);
+
+        Ok(())
+    }
+
+    pub fn output_inner(&mut self, value: JsValue, address: JsValue) -> WasmResult<()> {
+        let value_proto: Value = serde_wasm_bindgen::from_value(value)?;
+        let address_proto: Address = serde_wasm_bindgen::from_value(address)?;
+
+        self.planner
+            .output(value_proto.try_into()?, address_proto.try_into()?);
+
+        Ok(())
+    }
+
+    pub async fn swap_claim_inner(&mut self, swap_commitment: JsValue) -> WasmResult<()> {
+        let swap_commitment_proto: StateCommitment =
+            serde_wasm_bindgen::from_value(swap_commitment)?;
+
+        let swap_record: SwapRecord = self
+            .storage
+            .get_swap_by_commitment(swap_commitment_proto)
+            .await?
+            .expect("Swap record not found")
```

Comment:
If you are doing .expect() on these, why not just have `get_swap_by_commitment` return a Result instead of an Option?